File: UD-04/P02-Encordados-MongoDB-JPA-Java/src/main/java/es/joseluisgs/p02encordadosmongodbjpa/repositories/TenistasRepository.java
package es.joseluisgs.p02encordadosmongodbjpa.repositories;

import es.joseluisgs.p02encordadosmongodbjpa.models.Tenista;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TenistasRepository extends MongoRepository<Tenista, ObjectId> {
    Tenista findByNombre(String nombre);
}

File: UD-04/P03-Encordados-MongoDB-JPA-Kotlin/src/main/kotlin/es/joseluisgs/p02encordadosmongodbjpakotlin/repositories/TenistasRepository.kt
package es.joseluisgs.p02encordadosmongodbjpakotlin.repositories

import es.joseluisgs.p02encordadosmongodbjpakotlin.models.Tenista
import org.bson.types.ObjectId
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface TenistasRepository : MongoRepository<Tenista, ObjectId> {
    fun findByNombre(nombre: String): Tenista?
}